### PR TITLE
Add gear submission tests

### LINKED ISSUE #492: Only allow valid gears to be submitted to POST /gears
Now that we're using the `gear_tools` library, we can validate that a gear manifest is actually valid :+1:

### PR DESCRIPTION
**Pull request by kofalt**
Closes #492.

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by kofalt** [comment]
I don't have time to fix this test before I head out today, but everything should be basically ready.

Ping @ryansanford for acceptance as this affects the infrastructure.

**Comment by coltonlw** [comment]
@kofalt There are two tests failing, the first is the one that adds the dcm-convert gear.  I opened postman and executed that request against my local instance and got the response: 

```json
{
  "reason": "Gear manifest does not match schema",
  "key": null,
  "error": "'author' is a required property"
}
```

The second test is failing because it depends on that gear being added

**Comment by kofalt** [comment]
CI is green :tada:

File: api/jobs/handlers.py [hunk 1]
```diff
@@ -74,7 +74,20 @@ def post(self, _id):
...
 
-        upsert_gear(self.request.json)
+        try:
+            upsert_gear(self.request.json)
+        except ValidationError as err:
+            key = None
+            if len(err.relative_path) > 0:
+                key = err.relative_path[0]
+
+            self.response.set_status(400)
```
Comment (nagem) [inline api/jobs/handlers.py]: You use a 422 for a schema validation error in jobs/add - we should keep it consistent.

**Comment by nagem** [comment]
Other than the comment about the error code, LGTM.

File: api/jobs/handlers.py (hunk 1 above)
Comment (kofalt) [inline api/jobs/handlers.py]: Colton requested this be 400 for now until we go over this module with a full-verification pass :)

**Comment by kofalt** [comment]
Merging this based on our conversation

**Pull request merged**